app/models/letter.rb

```diff
@@ -8,8 +8,22 @@ class Letter
 
   attr_accessor :letter_template_id
 
-  def initialize(letter_template_id)
+  def initialize(letter_template_id, kase = nil)
     @letter_template_id = letter_template_id
+    @case = kase
+    @letter_template = LetterTemplate.find_by_id(letter_template_id)
+  end
+
+  def body
+    @letter_template&.render(@case)
+  end
+
+  def values
+    @case
+  end
+
+  def letter_date
+    Date.today.strftime('%e %b %Y')
   end
 end
 
```

Comment:
Regarding the initialize method signature... Can now pass in an optional Case instance. Annoyingly this still takes an integer for letter template ID - would prefer it took a LetterTemplate instance but that breaks how this is used as a form object when building a new letter, so left as-is for this iteration. In future iteration might split out an actual form object for that role and make this Letter class depend on a valid LetterTemplate

Reply:
I chose not to explicitly delegate named methods through to the case - instead just expose the case as "values" for the letter. This is because I don't want Letter to know so much about Case.